File: Q405.py
class Solution(object):
    def toHex(self, num):
        x=""
        if num==0:
            return "0"
        
        if num<0:
            num=num+pow(2,32)
        
        while num>0:
            a=num%16
            if a>9:
                x+=chr(a+87)
            else :
                x+=str(a)
            num=num//16 
        
        return x[::-1]

File: Q405.cpp
class Solution {
public:
    char digtochar(int x){
        if(x<10) return (char) ('0'+x);
        else return (char) ('a' + (x-10));
    }
    string toHex(int n) {
        if(n==0) return "0";
        unsigned int num = n;
        cout<<num;
        string res;
        while(num){
            res.push_back(digtochar(num%16));
            num/=16;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};


